### PR TITLE
Remove legacy Ray code.

### LINKED ISSUE #3128: Remove legacy Ray code.

### PR DESCRIPTION
**Pull request by robertnishihara**
This removes the old backend. This is a huge code change.

Currently, this PR removes all the Windows shims because they were under `src/common`. We'll presumably need to bring some of these back to support Windows, though perhaps it won't be necessary now that we're using asio instead of using sockets directly. We could also leave these in the repository in a different location. Any thoughts @mehrdadn?

**TODO:**
- [x] Need to change back to `apache/arrow` from `robertnishihara/arrow`.

This fixes #3128.

### CODE REVIEW

**Comment by chuxi** [comment]
Great Job!

**Comment by guoyuhong** [comment]
That's big milestone!

**Comment by atumanov** [comment]
love this, thanks for doing it, @robertnishihara ! 40K lines of removed code and counting!

**Comment by robertnishihara** [comment]
@mehrdadn sounds good, yeah that should work.

**Comment by robertnishihara** [comment]
jenkins, retest this please

File: src/ray/raylet/local_scheduler_client.cc [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+                fbb.GetSize(), fbb.GetBufferPointer(), &conn->write_mutex);
+  /* TODO(swang): Propagate the error. */
+}
+
+void local_scheduler_notify_unblocked(LocalSchedulerConnection *conn) {
+  write_message(conn->conn, static_cast<int64_t>(MessageType::NotifyUnblocked), 0, NULL,
+                &conn->write_mutex);
+}
+
+// const std::vector<uint8_t> local_scheduler_get_actor_frontier(
```
Comment (pcmoritz) [inline src/ray/raylet/local_scheduler_client.cc]: can you remove this?

File: src/ray/raylet/lib/python/local_scheduler_extension.cc [hunk 2]
```diff
@@ -208,34 +155,34 @@ static PyObject *PyLocalSchedulerClient_resource_ids(PyObject *self) {
...
-
-  std::vector<uint8_t> frontier;
-  Py_ssize_t length = PyBytes_Size(py_frontier);
-  char *frontier_data = PyBytes_AsString(py_frontier);
-  frontier.assign(frontier_data, frontier_data + length);
-  local_scheduler_set_actor_frontier(
-      ((PyLocalSchedulerClient *) self)->local_scheduler_connection, frontier);
-  Py_RETURN_NONE;
-}
+// static PyObject *PyLocalSchedulerClient_get_actor_frontier(PyObject *self,
```
Comment (pcmoritz) [inline src/ray/raylet/lib/python/local_scheduler_extension.cc]: can you remove this?

File: src/ray/raylet/lib/python/local_scheduler_extension.cc [hunk 3]
```diff
@@ -446,88 +381,82 @@ static PyObject *PyLocalSchedulerClient_free(PyObject *self, PyObject *args) {
...
-    {"resource_ids", (PyCFunction) PyLocalSchedulerClient_resource_ids,
-     METH_NOARGS,
+    {"resource_ids", (PyCFunction)PyLocalSchedulerClient_resource_ids, METH_NOARGS,
      "Get the IDs of the resources that are reserved for this client."},
-    {"get_actor_frontier",
-     (PyCFunction) PyLocalSchedulerClient_get_actor_frontier, METH_VARARGS, ""},
-    {"set_actor_frontier",
-     (PyCFunction) PyLocalSchedulerClient_set_actor_frontier, METH_VARARGS, ""},
-    {"wait", (PyCFunction) PyLocalSchedulerClient_wait, METH_VARARGS,
+    // {"get_actor_frontier",
```
Comment (pcmoritz) [inline src/ray/raylet/lib/python/local_scheduler_extension.cc]: remove this

File: src/ray/gcs/tables.h [hunk 4]
```diff
@@ -434,90 +429,91 @@ class TaskTable : public Table<TaskID, ray::protocol::Task> {
...
-};
-
-Status TaskTableAdd(AsyncGcsClient *gcs_client, Task *task);
-
-Status TaskTableTestAndUpdate(AsyncGcsClient *gcs_client, const TaskID &task_id,
-                              const ClientID &local_scheduler_id,
-                              SchedulingState test_state_bitmask,
-                              SchedulingState update_state,
-                              const TaskTable::TestAndUpdateCallback &callback);
+// class TaskTable : public Table<TaskID, TaskTableData> {
```
Comment (pcmoritz) [inline src/ray/gcs/tables.h]: remove this

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**